addons/http_routing/models/res_lang.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class Lang(models.Model):
+    _inherit = "res.lang"
+
+    url_code = fields.Char('URL Code', required=True, help='The Lang Code displayed in the URL')
+
+    _sql_constraints = [('url_code_uniq', 'unique(url_code)', 'This URL Code is already used.')]
+
+    def create(vals):
+        # TODO (review): Sadly can't have a `default=` using another self value as empty record set
+        if 'url_code' not in vals:
+            vals['url_code'] = vals.get('iso_code')
```

Comment:
if `iso_code` is required no need to use `get` (which implies `url_code` could be `None` which would be problematic)

Reply:
not required, was fixed locally, indeed if not iso_code should init on code